### ISSUE TITLE
Error when adding number values to `flexGrow` utilities in Tailwind config

### ISSUE THREAD

**Issue by brandonmcconnell** [tailwindlabs/tailwindcss#10404]
**What version of Tailwind CSS are you using?**

v3.2.4

**What build tool (or framework if it abstracts the build tool) are you using?**

Tailwind Play (but it's an issue with Tailwind's types, not Play specifically AFAICT)

**What version of Node.js are you using?**

N/A

**What browser are you using?**

Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

https://play.tailwindcss.com/irO8b9PPm1?file=config

**Describe your issue**

The Tailwind Play alerts an error when adding a numeric value to either `theme.flexGrow` or `theme.extend.flexGrow`, though it's exactly what the documentation recommends here: https://tailwindcss.com/docs/flex-grow#customizing-your-theme

This is likely the case for other properties as well. The right solution might be to add a new type to `interface ThemeConfig` (`types/config.d.ts`) to support key-numeric-value pairs like this. Something like…

```ts
type KeyValuePair<K extends keyof any = string, V = string> = Record<K, V>
type KeyValuePairNumeric<K extends keyof any = string, V = number> = Record<K, V>
```

**Comment by reinink** [comment]
Hey @brandonmcconnell! 👋

So good catch on the documentation. The docs should probably always quote these numeric values, since technically it's typed as a string, and if you look at our [default config](https://github.com/tailwindlabs/tailwindcss/blob/master/stubs/defaultConfig.stub.js) we use strings for all config values. I've just updated the docs to reflect this: https://github.com/tailwindlabs/tailwindcss.com/commit/af333c3e32a4e638d1d6d1d5ec9b3dcc97340f90

Using strings is quite often necessary since many values include units (ie. `px` and `rem`), so we've decided to just keep things simple in the config and make all config values strings.

That said, technically it will still work if you use numbers for config values — you'll just get type warnings. So if you don't want those warnings just use strings.

Hope that helps! 👍
